atomic_reactor/plugins/add_filesystem.py

```diff
@@ -343,17 +344,14 @@ def _add_filesystem_to_dockerfile(self, file_name, build_dir: BuildDir):
                 offset += 1
 
         build_dir.dockerfile.lines = lines
-        new_parents = []
-
-        for image in build_dir.dockerfile.parent_images:
-            if base_image_is_custom(image):
-                new_parents.append('scratch')
-            else:
-                new_parents.append(image)
-
-        build_dir.dockerfile.parent_images = new_parents
```

Comment:
or maybe not, does some other plugin do this change later?

Reply:
Should be no. There should be two plugins modifying `FROM`, `check_base_image` and `add_filesystem`.

I even thought whether we still need `change_from_in_df` plugin. If one plugin has to update `FROM` with a Dockerfile, that plugin should just do it, not delay to another plugin. Otherwise, the whole update logic might look like fragmented.

Reply:
Hmm, but this PR, `check_base_image` and `add_filesystem` would modify data.dockerfile_images in memory and `change_from_in_df` would later modify the Dockerfile on disk right?

I agree with you that updating the Dockerfile should just be done right away, not deferred to another plugin. In fact, I don't really understand why we have the in-memory representation of dockerfile_images at all :thinking: